Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -913,12 +1017,34 @@ extension DSLTree.Node {
     case .consumer, .matcher:
       // Allow zero width consumers and matchers
      return false
-    case .customCharacterClass:
-      return true
+    case .customCharacterClass(let ccc):
+      return ccc.guaranteesForwardProgress
     case .quantification(let amount, _, let child):
       let (atLeast, _) = amount.ast.bounds
       return atLeast ?? 0 > 0 && child.guaranteesForwardProgress
     default: return false
     }
   }
 }
+
+extension DSLTree.CustomCharacterClass {
+  /// We allow trivia into CustomCharacterClass, which could result in a CCC that matches nothing
+  /// ie (?x)[ ]
+  var guaranteesForwardProgress: Bool {
```

Comment:
I believe we should reject `(?x)[ ]` at parse time (we should reject a character class with no semantic members). That being said, it is possible to have a custom character class that matches nothing, e.g `[a--a]` (though as I understand it, that would still guarantee forward progress).